python/private/pypi/dependency_specifier_flag.bzl

```diff
@@ -0,0 +1,84 @@
+load("//python/private:toolchain_types.bzl", "TARGET_TOOLCHAIN_TYPE")
+load(":pep508_evaluate.bzl", "evaluate")
+
+def _impl(ctx):
+    env = {}
+
+    runtime = ctx.toolchains[TARGET_TOOLCHAIN_TYPE].py3_runtime
+    if runtime.interpreter_version_info:
+        version_info = runtime.interpreter_version_info
+        env["python_version"] = "{v.major}.{v.minor}".format(v = version_info)
+        full_version = format_full_version(version_info)
+        env["python_full_version"] = full_version
+        env["implementation_version"] = full_version
+    else:
+        env["python_version"] = get_flag(ctx.attr._python_version)
+        full_version = get_flag(ctx.attr._python_full_version)
+        env["python_full_version"] = full_version
+        env["implementation_version"] = full_version
+
+    # We assume cpython if the toolchain doesn't specify because it's most
+    # likely to be true.
+    env["implementation_name"] = runtime.implementation_name or "cpython"
+
+    env["os_name"] = struct()
```

Comment:
We can have a select statement in that sets a string based on `@platform//os:` or the user can override it. Have the default based on some logic could help.

Reply:
Done.

(aside: this value is kinda weird, see https://github.com/bazel-contrib/rules_python/issues/2826#issuecomment-2833578913 )